### ISSUE TITLE
Missing scrollbar on latest Debian's Firefox ESR

### ISSUE THREAD

**Issue by alkisg** [squidfunk/mkdocs-material#3346]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

On the latest Debian stable with [Firefox ESR 78.14](https://packages.debian.org/bullseye/firefox-esr), the sidebar doesn't show any scrollbars.

This means that users can't see all of our menu and can't navigate properly.

### Expected behaviour

A scrollbar should appear.

### Actual behaviour

A scrollbar doesn't appear.

### Steps to reproduce

1. Install the latest Debian Bullseye
2. Visit [this site](https://pe86.github.io/linux/ltsp/swap-partition/) with Firefox
3. Try to see all the menu at the left; it's not possible.

### Package versions

- Python: `3.9.9`
- MkDocs: `1.2.3`
- Material: `8.1.0`

### Configuration

```yaml
-
```

### System information

- Operating system: Debian Bullseye
- Browser: Firefox ESR

Attaching a screenshot where you can see that the scrollbar doesn't exist.

Note that in the screenshot, a **second** issue appears, that rarely happens: when scrolling up with the mouse "too much", something malfunctions and the bottom bar gets scrolled. I should probably file a different issue for this, but I thought I'd upload the screenshot now as it's rare to reproduce it...

![VirtualBox_bullseye-kde_13_12_2021_12_30_17](https://user-images.githubusercontent.com/88025/145797288-8397e5e9-eae8-4a5d-859c-9285e2a3f6eb.png)

**Comment by squidfunk** [comment]
Thanks for the screenshot. My guess is that the scrollbar is not the problem. The problem is that there's likely some error in the console that will hinder the JavaScript from mounting the scrollbar. Please check the console and see whether there's anything there that helps to understand the source of error.

**Comment by alkisg** [comment]
You are correct. The console error message is:

> Uncaught DOMException: Element.closest: ':not(td.code) > .highlight, .highlighttable' is not a valid selector

**Comment by alkisg** [comment]
The problem seems to be the `:not` CSS operator, it needs very very recent browsers:
https://developer.mozilla.org/en-US/docs/Web/CSS/:not

I.e. a plain `document.querySelector(":not(td.code)")` raises the exception.

**Comment by squidfunk** [comment]
Okay, one final idea – we might split it into the super ugly `:not(td):not(.code)`. Does that work in the console?

**Comment by alkisg** [comment]
Yes, indeed it works!

**Comment by squidfunk** [comment]
Then try to patch the `bundle.******.js` file first, checking if there are any other obstacles. If there aren't, we can change that line.

**Comment by alkisg** [comment]
After the following command, everything seems fine:

```shell
sed 's/:not(td.code)/:not(td):not(.code)/' -i assets/javascripts/bundle.*.min.js
```

**Comment by squidfunk** [comment]
Ok great, we can integrate this into the code base, as it's a simple fix.

**Comment by squidfunk** [comment]
Fixed in dea3ad86c.

**Comment by squidfunk** [comment]
Released as part of 8.1.1.
